tests/rkt_export_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/coreos/rkt/common"
+	"github.com/coreos/rkt/tests/testutils"
+)
+
+func TestExport(t *testing.T) {
+	const (
+		testAci     = "test.aci"
+		testFile    = "test.txt"
+		testContent = "ThisIsATest"
+	)
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	type testCfg struct {
+		runArgs        string
+		writeArgs      string
+		readArgs       string
+		expectedResult string
+	}
+
+	tests := []testCfg{
+		{
+			"--no-overlay --insecure-options=image",
+			"--write-file --file-name=" + testFile + " --content=" + testContent,
+			"--read-file --file-name=" + testFile,
+			testContent,
+		},
+	}
+
+	// Need to do both checks
+	if common.SupportsUserNS() && checkUserNS() == nil {
+		tests = append(tests, []testCfg{
+			{
+				"--private-users --no-overlay --insecure-options=image",
+				"--write-file --file-name=" + testFile + " --content=" + testContent,
+				"--read-file --file-name=" + testFile,
+				testContent,
+			},
+		}...)
+	}
+
+	if common.SupportsOverlay() {
+		tests = append(tests, []testCfg{
+			{
+				"--insecure-options=image",
+				"--write-file --file-name=" + testFile + " --content=" + testContent,
+				"--read-file --file-name=" + testFile,
+				testContent,
+			},
+		}...)
+	}
+
+	for _, tt := range tests {
+		tmpDir := createTempDirOrPanic("rkt-TestExport-tmp-")
+		defer os.RemoveAll(tmpDir)
+
+		tmpTestAci := filepath.Join(tmpDir, testAci)
+
+		// Prepare the image with modifications
+		const runInspect = "%s %s %s %s --exec=/inspect -- %s"
+		prepareCmd := fmt.Sprintf(runInspect, ctx.Cmd(), "prepare", tt.runArgs, getInspectImagePath(), tt.writeArgs)
+		t.Logf("Preparing 'inspect --write-file'")
+		uuid := runRktAndGetUUID(t, prepareCmd)
+
+		runCmd := fmt.Sprintf("%s run-prepared %s", ctx.Cmd(), uuid)
+		t.Logf("Running 'inspect --write-file'")
+		child := spawnOrFail(t, runCmd)
+		waitOrFail(t, child, 0)
+
+		// Export the image
```

Comment:
Can we run an additional test that umount the overlay fs manually before exporting the container?